tests/testsuite/freshness_checksum.rs

```diff
@@ -0,0 +1,2880 @@
+//! Tests for fingerprinting (rebuild detection).
+
+#![allow(deprecated)]
+
+use std::fs::{self, OpenOptions};
+use std::io::prelude::*;
+use std::net::TcpListener;
+use std::process::Stdio;
+use std::thread;
+
+use super::death;
+use cargo_test_support::paths::CargoPathExt;
+use cargo_test_support::prelude::*;
+use cargo_test_support::registry::Package;
+use cargo_test_support::{basic_lib_manifest, basic_manifest, project, rustc_host, rustc_host_env};
+
+#[cargo_test]
+fn checksum_actually_uses_checksum() {
+    let p = project()
+        .file("src/main.rs", "mod a; fn main() {}")
+        .file("src/a.rs", "")
+        .build();
+
+    p.cargo("build")
```

Comment:
Is there a reason we need `cargo build` instead of `cargo check`?  The latter helps keep test times down

Reply:
No, where it was possible to do so without breaking tests I downgraded these to `cargo check` in 3cc6814ee8ede4ab4f3defb581f37c0ae1a1623a